### PR TITLE
Document catch all routes

### PR DESCRIPTION
**Pull request by ereslibre**
Fixes: https://github.com/vmware-labs/wasm-workers-server/issues/213

### CODE REVIEW

File: docs/docs/features/dynamic-routes.md [hunk 1]
```diff
@@ -56,13 +56,61 @@ $ tree .
...
+│   └── [...slug]
+│       └── index.js
+└── [...slug]
+    └── index.js
+```
+
+Here, we have the same structure as in the previous example, but we have two catch-all, under two different roots:
+
+- One catch-all, `[...slug]/index.js` is serving all requests, except for requests whose path starts with `/about` or `/other`.
+- Another catch-all, `other/[...slug]/index.js` serves all requests under the `/other` path.
```
Comment (Angelmmiguel) [inline docs/docs/features/dynamic-routes.md:101]: This is great @ereslibre ! I would add some examples in which the segments are defined in the file directly (`[...slug].js`).

File: docs/docs/features/dynamic-routes.md [hunk 2]
```diff
@@ -56,13 +56,61 @@ $ tree .
...
+
+### Routing priority
+
+Given catch-all routes could potentially shadow other routes, it is important to settle precedence when routing requests. The rule of thumb is more specific routes win. This is, a route with no catch-all will always against a route with catch-all when they are at the same depth.
```
Comment (Angelmmiguel) [inline docs/docs/features/dynamic-routes.md:105]: Minor suggestions:

```suggestion
Given catch-all routes could potentially shadow other routes, it is important to settle precedence when routing requests. **The rule of thumb is more specific routes win**. For example, a route with no catch-all will always against a route with catch-all when they are at the same depth.
```

File: examples/js-catchall/other/[...slug]/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (Angelmmiguel) [inline examples/js-catchall/other/[...slug]/index.js:1]: I would use a here a file directly: `examples/js-catchall/other/[...slug].js`

**Review by Angelmmiguel — APPROVED** [review]
LGTM in general! Added some minor comments, but approving it in advance so you can merge after fixing it :).

Thank you for fixing all the tables btw!

**Comment by ereslibre** [comment]
> Thank you for fixing all the tables btw!

Haha, my editor does automatically when I edit a table :D

File: examples/js-catchall/other/[...slug]/index.js (hunk 3 above)
Comment (ereslibre) [reply examples/js-catchall/other/[...slug]/index.js:1]: I can add the one you propose too, but thought that having two catch-all at the directory level to show precedence at that level is also nice and somewhat documenting.

I will add this one too!

Comment (ereslibre) [reply examples/js-catchall/other/[...slug]/index.js]: Ah nevermind I see what you meant here, you are right. Updated.

**Pull request merged**